backend/controller/dal/dal.go

```diff
@@ -552,7 +565,26 @@ func (d *DAL) KillStaleControllers(ctx context.Context, age time.Duration) (int6
 }
 
 // DeregisterRunner deregisters the given runner.
-func (d *DAL) DeregisterRunner(ctx context.Context, key model.RunnerKey) error {
+func (d *DAL) DeregisterRunner(ctx context.Context, key model.RunnerKey, deployment model.DeploymentKey) error {
+	prev, err := d.db.GetRunnersForDeployment(ctx, deployment)
```

Comment:
consider evaluating the error or renaming it to _ and providing some comment explaining why error evaluation isn't necessary